src/vos/vos_query.c

```diff
@@ -203,25 +198,58 @@ query_recx(struct open_query *query, daos_recx_t *recx)
 	if (rc != 0)
 		goto fini;
 
-	recx->rx_idx = entry.en_sel_ext.ex_lo;
-	recx->rx_nr = entry.en_sel_ext.ex_hi - entry.en_sel_ext.ex_lo + 1;
+	if (opc & EVT_ITER_SKIP_HOLES) {
+		if (entry.en_visibility & EVT_VISIBLE) {
+			recx->rx_idx = entry.en_sel_ext.ex_lo;
+			recx->rx_nr = entry.en_sel_ext.ex_hi - entry.en_sel_ext.ex_lo + 1;
+		} else {
+			rc = -DER_NONEXIST;
+		}
+		D_GOTO(fini, rc);
+	}
+
+	/* The following only supports MAX recx query at the moment */
+	if (!(query->qt_flags & VOS_GET_MAX))
+		D_GOTO(fini, rc);
```

Comment:
rc = -DER_INVAL ?  rc = -DER_NOSYS ?  Maybe check this earlier and just assert here like

if (stripe_size != 0 && (flags & VOS_GET_MAX) == 0)
   ...

Reply:
yeah, probably assertion.